autofaiss/external/optimize.py

```diff
@@ -478,13 +477,11 @@ def optimize_and_measure_indices(
     indices_file_paths = fs.ls(indices_folder, detail=False)
     index_path2_metric_infos: Dict[str, Dict] = {}
     suffix_width = int(math.log10(len(indices_file_paths))) + 1
-    with tempfile.TemporaryDirectory() as local_indices_folder:
-        parallel_download_indices_from_remote(
-            fs=fs, indices_file_paths=indices_file_paths, dst_folder=local_indices_folder
-        )
-        for i, tmp_filename in enumerate(sorted(os.listdir(local_indices_folder)), 1):
-            index_filepath = os.path.join(local_indices_folder, tmp_filename)
-            index = faiss.read_index(index_filepath)
+    for i, index_file_path in enumerate(sorted(indices_file_paths), 1):
+        with tempfile.NamedTemporaryFile() as local_index_file:
+            local_index_filename = local_index_file.name
+            download_one(src_dst_path=(index_file_path, local_index_filename), fs=fs)
+            index = faiss.read_index(local_index_filename)
```

Comment:
you can even use fsspec to directly load it in memory without a local copy like this https://github.com/criteo/autofaiss/blob/6a03650ff7dda47359f6e7d1f731829dac346a66/autofaiss/external/quantize.py#L373

Reply:
Yes, good idea. Why not !